sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsCheckpointMark.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.jms;
+
+import org.apache.beam.sdk.coders.DefaultCoder;
+import org.apache.beam.sdk.coders.SerializableCoder;
+import org.apache.beam.sdk.io.UnboundedSource;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+/**
+ * Checkpoint for an unbounded JmsIO.Read. Consists of
+ * JMS destination name, and the latest message ID consumed so far.
+ */
+@DefaultCoder(SerializableCoder.class)
```

Comment:
`SerializableCoder` is *strongly* discouraged in streaming pipelines / unbounded PCollections. It is very verbose (so that checkpoints and records are larger, increasing latency and the amount of data that has to be committed). It is also not guaranteed to be deserializable to an equal object in a different version of the JVM, so you might lose data if you restart a pipeline with an updated version of the code.